src/components/Calculator.js

```diff
@@ -1,30 +1,47 @@
-import React from 'react';
+import React, { Component } from 'react';
+import Button from './Button';
+import Display from './Display';
+import calculate from '../logic/calculate';
 import './calculator.css';
 
-class Calculator extends React.Component {
+class Calculator extends Component {
+  constructor() {
+    super();
+    this.state = {
+      total: '0',
+      next: null,
+      operation: null,
+    };
+  }
+
+  handleClick = (val) => {
+    this.setState((prevState) => calculate(prevState, val));
+  }
+
   render() {
+    const { total, next, operation } = this.state;
     return (
       <div className="calculator-grid">
-        <div className="output">0</div>
-        <button>AC</button>
-        <button>+/-</button>
-        <button>%</button>
-        <button className="bg-secondary">÷</button>
-        <button>7</button>
-        <button>8</button>
-        <button>9</button>
-        <button className="bg-secondary">x</button>
-        <button>4</button>
-        <button>5</button>
-        <button>6</button>
-        <button className="bg-secondary">-</button>
-        <button>1</button>
-        <button>2</button>
-        <button>3</button>
-        <button className="bg-secondary">+</button>
-        <button className="span-two">0</button>
-        <button>.</button>
-        <button className="bg-secondary">=</button>
+        <Display css="output" total={total} next={next} operation={operation} />
+        <Button buttonName="AC" handleClick={this.handleClick} />
+        <Button buttonName="+/-" handleClick={this.handleClick} />
+        <Button buttonName="%" handleClick={this.handleClick} />
+        <Button buttonName="÷" css="bg-secondary" handleClick={this.handleClick} />
+        <Button buttonName="7" handleClick={this.handleClick} />
+        <Button buttonName="8" handleClick={this.handleClick} />
+        <Button buttonName="9" handleClick={this.handleClick} />
+        <Button buttonName="x" css="bg-secondary" handleClick={this.handleClick} />
+        <Button buttonName="4" handleClick={this.handleClick} />
+        <Button buttonName="5" handleClick={this.handleClick} />
+        <Button buttonName="6" handleClick={this.handleClick} />
+        <Button buttonName="-" css="bg-secondary" handleClick={this.handleClick} />
+        <Button buttonName="1" handleClick={this.handleClick} />
+        <Button buttonName="2" handleClick={this.handleClick} />
+        <Button buttonName="3" handleClick={this.handleClick} />
+        <Button buttonName="+" css="bg-secondary" handleClick={this.handleClick} />
+        <Button buttonName="0" css="span-two" handleClick={this.handleClick} />
+        <Button buttonName="." handleClick={this.handleClick} />
+        <Button buttonName="=" css="bg-secondary" handleClick={this.handleClick} />
       </div>
```

Comment:
- [x] OPTIONAL: This could have been a separate component (`Buttons`) in a separate file and rendered here. Smaller components are easy to fix, reusable, and make the app more scalable. :pray: